### DISCUSSION TITLE
How to use Wait?

### DISCUSSION THREAD

**Discussion by RazortDev** [scripthookvdotnet/scripthookvdotnet#1320] [Q&A]
`using GTA;
using GTA.Math;
using GTA.Native;
using iFruitAddon2;
using System;
using System.Windows.Forms;
using System.Threading;

public class HackingMission : Script
{
    private int wait = -1;
    CustomiFruit _iFruit;
    private int rewardAmount;
    public void SetWait(int ms)
    {
        if (ms > wait)
        {
            wait = ms;
        }
    }

    public HackingMission()
    {
        _iFruit = new CustomiFruit();

        iFruitContact contactA = new iFruitContact("Kai");
        contactA.Answered += ContactAnswered;   // Linking the Answered event with our function
        contactA.DialTimeout = 1500;            // Delay before answering
        contactA.Active = true;                 // true = the contact is available and will answer the phone
        contactA.Icon = ContactIcon.Blank;      // Contact's icon
        _iFruit.Contacts.Add(contactA);         // Add the contact to the phone

        Tick += OnTick;
    }

    private void OnTick(object sender, EventArgs e)
    {
        if (wait > -1)
        {
            Wait(wait);
            wait = -1;
        }
        _iFruit.Update();
    }

    private void ContactAnswered(iFruitContact contact)
    {
        _iFruit.Close(1000);

        int chance = new Random().Next(0, 100);
        int security = new Random().Next(0, 100);
        string securitytier = "Not Setted";
        int finalchance = chance - security;

        if(security > 0) { securitytier = "~g~ Low"; } else
        if (security > 30) { securitytier = "~y~ Middle"; } else
        if (security > 70) { securitytier = "~r~ High"; };

        GTA.UI.Notification.Show(GTA.UI.NotificationIcon.BlankEntry, "Kai", "Website Information", "You got website with " + securitytier + "~s~ Security", false, false);
        SetWait(5000);
        GTA.UI.Screen.ShowHelpText("You trying to hack the website...", 6000);
        SetWait(5500);
        if (finalchance > 50)
        {
            Vector3 playerPosition = Game.Player.Character.Position;

            // Spawn police vehicles nearby
            Vehicle[] policeVehicles = new Vehicle[3];
            for (int i = 0; i < policeVehicles.Length; i++)
            {
                policeVehicles[i] = World.CreateVehicle(VehicleHash.Police, playerPosition.Around(30f));
                policeVehicles[i].IsPersistent = true;
            }
            GTA.UI.Notification.Show(GTA.UI.NotificationIcon.Call911, "Police Department", "~r~ Track Your Location", "~s~We get a report if you have attempted to cross, we will immediately find your location.", false, false);
            SetWait(5000);
            Function.Call(Hash.SET_PLAYER_WANTED_LEVEL, Game.Player.Character.Handle, 3, true);
        } else if(finalchance < 51)
        {
            rewardAmount = security * 200;
            GTA.UI.Notification.Show(GTA.UI.NotificationIcon.BlankEntry, "Kai", "Success", "You did it! You got ~g~"+ rewardAmount +"$", false, false);
            Game.Player.Character.Money += rewardAmount;
        }
    }
}`

the SetWait doesn't work, help me

**Comment by kagikn** [comment]
`Script.Wait` makes the script wait for at least a given milliseconds, but your `SetWait` just sets a int to a private field. But the way I see it, I think you should call `Update` on the iFruit instance every frame and not call `Script.Wait` (before calling `Update` on the iFruit instance). Try splitting your phone answered code into smaller pieces.
